src/odemis/gui/cont/tabs.py

```diff
@@ -2417,48 +2472,151 @@ def _update_progress_bar(self, pos):
+                    # initially, choose grid1 by default 
+                    if currently_pressed.Label == meteor_labels[FM_IMAGING]:
+                        self.position_btns[FM_IMAGING].SetValue(True)
+                        self.position_btns[GRID_1].Enable()
+                        self.position_btns[GRID_1].SetValue(True)
+                    elif currently_pressed.Label == meteor_labels[SEM_IMAGING]:
+                        self.position_btns[SEM_IMAGING].SetValue(True) 
+                        self.position_btns[GRID_1].Enable()
+                        self.position_btns[GRID_1].SetValue(True)
+                else: # if user closed the gui, and opened it again
+                    current_pos_label = getCurrentPositionLabel(self._stage.position.value)
+                    current_grid_label = getCurrentGridLabel(self._stage.position.value)
+                    # if grid already selected
+                    if current_grid_label == GRID_1:
+                        self.position_btns[GRID_1].SetValue(True)
+                        self.position_btns[GRID_2].SetValue(False)
+                    elif current_grid_label == GRID_2:
+                        self.position_btns[GRID_2].SetValue(True)
+                        self.position_btns[GRID_1].SetValue(False)
+                    if current_pos_label == SEM_IMAGING:
+                        self.position_btns[SEM_IMAGING].SetValue(True)
+                        self.position_btns[FM_IMAGING].SetValue(False)
+                    elif current_pos_label == FM_IMAGING:
+                        self.position_btns[SEM_IMAGING].SetValue(False)
+                        self.position_btns[FM_IMAGING].SetValue(True)
+            # if user already pressed a button, and now presses a new button
+            elif self.position_btns[GRID_2].GetValue() == 1 or self.position_btns[GRID_1].GetValue() == 1:
+                # if user switches grids, keep imaging modes same. 
+                if currently_pressed.Label == meteor_labels[GRID_1]:
+                    self.position_btns[GRID_1].SetValue(True)
+                    self.position_btns[GRID_2].SetValue(False)
+                # if user switches grids, keep imaging modes same.
+                elif currently_pressed.Label == meteor_labels[GRID_2]:
+                    self.position_btns[GRID_2].SetValue(True)
+                    self.position_btns[GRID_1].SetValue(False)
+                # if user switches imaging modes, keep grids the same. 
+                elif currently_pressed.Label == meteor_labels[SEM_IMAGING]:
+                    self.position_btns[SEM_IMAGING].SetValue(True)
+                    self.position_btns[FM_IMAGING].SetValue(False)
+                # if user switches imaging modes, keep grids the same. 
+                elif currently_pressed.Label == meteor_labels[FM_IMAGING]:
+                    self.position_btns[SEM_IMAGING].SetValue(False)
+                    self.position_btns[FM_IMAGING].SetValue(True)
```

Comment:
It can not replace all the code. We have to detect and handle the initial case separately as it is different from the normal cases of pressing a button. But a part of the code seemed not needed after i had another look. So now it is less complicated. However, note that with this implementation if you press a button, it will not be immediately toggled, and stay toggled during the whole move. For example, assume currently the stage is in the sem area. If we press fm button, then it will not toggle fm button directly, because the stage has not totally moved out of the sem area yet. It will take some time until to goes out the sem area, then it will toggle the fm button. I have tried hard to make it similar to enzel buttons behavior, but i think it can not be done in the same way because of the listener of the stage position va. I could handle the toggling properly in this function using the button object itself, not the labels, and then it can be similar to enzel buttons behavior. But in the listener we do not have the button object, and so we have to use the labels which means that we can not have full control on how and when the buttons are toggled (as i mentioned mentioned in the example above).